h5pyd/_apps/utillib.py

```diff
@@ -1105,15 +1104,25 @@ def create_dataset(dobj, ctx):
                     new_chunks = [1,]
                     new_chunks.extend(chunks)
                     chunks = tuple(new_chunks)
+                    logging.debug("extend chunks for preappend:", chunks)
             else:
                 if isinstance(chunks, dict):
                     if "dims" in chunks:
                         chunk_dims = chunks["dims"]
-                        if len(chunk_dims) == 1:
-                            # currently hyperchunks only supported for 1d datasets
-                            chunk_dims = expandChunk(chunk_dims, dobj.shape, dobj.dtype.itemsize)
-                            logging.debug(f"expanded chunks: {chunk_dims}")
-                            chunks["dims"] = chunk_dims
+                        layout_class = chunks.get("class")
+                        server_version = fout.serverver
+                        if server_version and server_version.startswith("0.9"):
```

Comment:
This version check should probably be implemented as a greater-than comparison on tuples so it doesn't break if HSDS switches to a `1.X.X` version